### PR TITLE
Pod limits, cache service and Go 1.13 compatibility

### PR DESCRIPTION
**Pull request by galderz**
Several pieces of work:

* Set pod cpu/memory limits (#205), this is required to build the default cache used by cache service (#134).
* Add cache service support (#134).
* Update code to StatefulSet `app/v1` and make it compile/test on Go 1.13.

### CODE REVIEW

**Comment by galderz** [comment]
@rigazilla Squashed commits. If CI passes this is ready to go in.

File: pkg/controller/infinispan/util/cluster.go [hunk 1]
```diff
@@ -90,5 +92,96 @@ func (c Cluster) GetClusterMembers(secretName, podName, namespace, protocol stri
...
+	execOut, execErr, err := c.Kubernetes.ExecWithOptions(execOptions)
+	if err != nil {
+		return fmt.Errorf("unexpected error creating cache, stderr: %v, err: %v", execErr, err)
+	}
+
+	// Reverse sort the standard output so that HTTP status code is first
+	execOutLines := strings.Split(execOut.String(), "\n")
+	reverse(execOutLines)
+
+	httpCode, err := strconv.ParseUint(execOutLines[0], 10, 64)
```
Comment (rigazilla) [inline pkg/controller/infinispan/util/cluster.go]: Not a bug but
seems you only use the last execOutLines string. Do you really need to reverse the array?

Comment (galderz) [reply pkg/controller/infinispan/util/cluster.go]: Yeah, that's on purpouse. The curl as defined shows the output and then a line break and the HTTP status code. So, to get the status code I split the entire output by line and reverse it. Then, the first element would be the HTTP status code alone.

Comment (galderz) [reply pkg/controller/infinispan/util/cluster.go]: The output of the command can itself have line breaks.

**Comment by rigazilla** [comment]
merged, thanks!

**Pull request merged**